xbmc/addons/PVRClient.cpp

```diff
@@ -591,25 +591,13 @@ PVR_ERROR CPVRClient::RenameChannel(const CPVRChannelPtr &channel)
   }, m_clientCapabilities.SupportsChannelSettings());
 }
 
-PVR_ERROR CPVRClient::GetEPGForChannel(const std::shared_ptr<CPVREpgChannelData>& channelData,
+PVR_ERROR CPVRClient::GetEPGForChannel(int iChannelUid,
                                        CPVREpg* epg,
                                        time_t start /* = 0 */,
                                        time_t end /* = 0 */,
                                        bool bSaveInDb /* = false */)
 {
-  return DoAddonCall(__FUNCTION__, [this, channelData, epg, start, end, bSaveInDb](const AddonInstance* addon) {
-
-    //! @todo PVR Addon API Change: Change GetEPGForChannel param from 'PVR_CHANNEL channel' to 'int iUniqueId'.
-    PVR_CHANNEL addonChannel = {0};
-
-    // mandatory
-    addonChannel.iUniqueId = channelData->UniqueClientChannelId();
-    addonChannel.bIsRadio = channelData->IsRadio();
```

Comment:
What about this field?

Reply:
Good catch. I guess this can be removed.

Reply:
Well, this is okay as it is. All the addon needs is the uid of the channel. Everything else can be derived from the uid by the adodn itself, including the "isradio" flag.

Reply:
True